rs/consensus/src/idkg/utils.rs

```diff
@@ -416,17 +416,22 @@ pub(crate) fn inspect_chain_key_initializations(
                 )
             })?;
 
-        let dealings = chain_key_init
-            .dealings
-            .as_ref()
-            .expect("Error: Failed to find dealings in chain_key_initializations")
-            .try_into()
-            .map_err(|err| {
-                format!(
-                    "Error reading initial IDkg dealings: {:?}. Setting idkg_summary to None.",
-                    err
+        let dealings = match &chain_key_init.initialization {
+            Some(pb::chain_key_initialization::Initialization::Dealings(dealings)) => dealings,
+            Some(pb::chain_key_initialization::Initialization::Transcript(_)) => continue,
+            None => {
+                return Err(
+                    "Error: Failed to find dealings in chain_key_initializations".to_string(),
                 )
-            })?;
+            }
+        };
+
+        let dealings = dealings.try_into().map_err(|err| {
+            format!(
+                "Error reading initial IDkg dealings: {:?}. Setting idkg_summary to None.",
```

Comment:
We can do this in a separate PR but as discussed, I think the correct behavior here and in other places of this function, is actually to panic if we can't deserialize the initializations properly. Otherwise, if we simply ignore the malformed initialization, then the subnet will instead generate a _new_ key for the key ID that should have been reshared. So then we would have two subnets with different keys for the same key ID, which we should avoid.

Panicking should be safe, because then we can simply propose a new, higher recovery CUP (with the same state hash), that doesn't cause a panic.

In addition, I think one improvement we should make in the orchestrator is to only call `make_registry_cup_from_cup_contents` if the height of those cup contents is either higher than the local CUP, or the local CUP is a registry CUP.

Reply:
Alternatively we should add a condition such that the replica will never try to create a new key for a key ID that exists in the ChainKeyConfig of multiple subnets according to the registry.

Reply:
Yes, we should handle this in a separate PR. I think it would be better to return errors here, and then in on the callsite panic on the error.